models/events.go

```diff
@@ -1,114 +1,142 @@
 package models
 
 import (
+	"fmt"
 	"strings"
 
 	"github.com/onflow/cadence"
 	"github.com/onflow/flow-go-sdk"
-	"github.com/onflow/flow-go/fvm/evm/types"
-	"golang.org/x/exp/slices"
+	"github.com/onflow/flow-go/fvm/evm/events"
 )
 
 // isBlockExecutedEvent checks whether the given event contains block executed data.
 func isBlockExecutedEvent(event cadence.Event) bool {
 	if event.EventType == nil {
 		return false
 	}
-	return strings.Contains(event.EventType.ID(), string(types.EventTypeBlockExecuted))
+	return strings.Contains(event.EventType.ID(), string(events.EventTypeBlockExecuted))
 }
 
 // isTransactionExecutedEvent checks whether the given event contains transaction executed data.
 func isTransactionExecutedEvent(event cadence.Event) bool {
 	if event.EventType == nil {
 		return false
 	}
-	return strings.Contains(event.EventType.ID(), string(types.EventTypeTransactionExecuted))
+	return strings.Contains(event.EventType.ID(), string(events.EventTypeTransactionExecuted))
 }
 
+// CadenceEvents contains Flow emitted events containing one or zero evm block executed event,
```

Comment:
```suggestion
// CadenceEvents contains Flow emitted events containing exactly one evm block executed event,
```

Reply:
we will leave an option for Flow block to not contain EVM block as per protocol design, we will now have 1 but in future we might not

Reply:
Ah, I see. I wasn't aware of that :+1: